python/modules/connection.py

```diff
@@ -175,133 +318,112 @@ class Connection(Module):
+            their_did,
+            json.dumps({
+                'label': label,
+                'endpoint': their_endpoint
+            })
         )
 
-        meta_json = json.dumps(
-            {
-                "conn_name": conn_name,
-                "their_endpoint": their_endpoint_uri,
-                "their_verkey": their_key_str,
-                "my_verkey": my_new_verkey
-            }
+        # Create pairwise relationship between my did and their did
+        await pairwise.create_pairwise(
+            self.agent.wallet_handle,
+            their_did,
+            my_did,
+            json.dumps({
+                'label': label,
+                'their_endpoint': their_endpoint,
+                'their_vk': their_vk,
+                'my_vk': my_new_vk
+            })
         )
 
-        await did.store_their_did(self.agent.wallet_handle, identity_json)
-        await pairwise.create_pairwise(self.agent.wallet_handle, their_did_str, my_did_str, meta_json)
-
         await self.agent.send_admin_message(
             Message({
                 '@type': AdminConnection.REQUEST_RECEIVED,
-                'content': {
-                    'name': conn_name,
-                    'endpoint_did': their_did_str,
-                    'history': msg
-                }
+                'label': label,
+                'did': their_did,
+                'endpoint': their_endpoint,
+                'history': msg
             })
         )
 
 
     async def response_received(self, msg: Message) -> Message:
-        my_did_str = msg.context['to_did']
-        their_did_str = msg['did']
-        their_key_str = msg.context['from_key']
-
-        my_did_info_str = await did.get_my_did_with_meta(self.agent.wallet_handle, my_did_str)
-        my_did_info_json = json.loads(my_did_info_str)
-
-        my_verkey = my_did_info_json['verkey']
-        metadata_str = my_did_info_json['metadata']
-        metadata_dict = json.loads(metadata_str)
-
-        conn_name = metadata_dict['conn_name']
-        their_endpoint = metadata_dict['their_endpoint']
-
-        identity_json = json.dumps(
-            {
-                "did": their_did_str,
-                "verkey": their_key_str
-            }
-        )
-
-        meta_json = json.dumps(
-            {
-                "conn_name": conn_name,
-                "their_endpoint": their_endpoint,
-                "their_verkey": their_key_str,
-                "my_verkey": my_verkey
-            }
+        """ Process response
+
+            Response format:
+                {
+                  "@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/connections/1.0/response",
+                  "DID":"A.did@A:B",
+                  "DIDDoc": {
+                      //did doc
+                  }
+                }
+        """
+        my_did = msg.context['to_did']
+        their_did = msg['DID']
+        their_new_vk = msg.context['from_key'] # equivalent to msg['DIDDoc']['key']?
+
```

Comment:
We need a comment in here indicating that we ignored signature checking for now, but this is the proper place to perform that check.